includes/multi-currency/RestController.php

```diff
@@ -67,4 +87,51 @@ public function update_enabled_currencies( $request ) {
 		WC_Payments_Multi_Currency()->set_enabled_currencies( $params['enabled'] );
 		return $this->get_store_currencies();
 	}
+
+	/**
+	 * Gets the currency settings for a single currency.
+	 *
+	 * @param   \WP_REST_Request $request  Full data about the request.
+	 *
+	 * @return  array            The currency settings.
+	 */
+	public function get_currency_settings( $request ) {
+		$currency_code = sanitize_key( strtolower( $request['currency_code'] ) );
+		return [
+			'exchange_rate_type' => get_option( 'wcpay_multi_currency_exchange_rate_' . $currency_code, 'automatic' ),
+			'manual_rate'        => get_option( 'wcpay_multi_currency_manual_rate_' . $currency_code, null ),
+			'price_rounding'     => get_option( 'wcpay_multi_currency_price_rounding_' . $currency_code, null ),
+			'price_charm'        => get_option( 'wcpay_multi_currency_price_charm_' . $currency_code, null ),
+		];
+	}
+
+
+	/**
+	 * Updates the currency settings for a single currency.
+	 *
+	 * @param   \WP_REST_Request $request  Full data about the request.
+	 *
+	 * @return  array            The currency settings.
+	 */
+	public function update_currency_settings( $request ) {
+		$currency_code = sanitize_key( strtolower( $request['currency_code'] ) );
+		$params        = $request->get_params();
+
+		if ( preg_match( '/^[a-z]{3}$/', $currency_code ) ) {
```

Comment:
I'm wondering if perhaps here it may be better to check the list of `available_currencies` from `MultiCurrency` to make sure the currency is valid, rather than validating that it is 3 characters? If it's not feasible to do that easily from within the context of the `RestController`, I think this approach wouldn't an issue, however.

Reply:
Thanks @dmallory42 for the detailed review, I knew you would make it 🚀 😁 Here are my answers:

>I noticed that clicking on the "Learn more" links in the Formatting Rules section currently has no effect (not sure if this is in the scope of the PR but thought I would mention):

I couldn't find any links for them, since the old version doesn't contain them and the new version I copied from Figma's that @LevinMedia provided, so I don't know where they should navigate the user.

> I noticed that when I change the currency rate (e.g. switch from automatic to manual and change the rate), save the change, and then go back to the main Enabled Currencies page via the breadcrumb link, the rate displayed in the main table containing all of the enabled currencies does not update for the currency. I have to refresh the page to get this rate to update. Is it possible to make this update once the change is saved on the single currency settings page?

Yep, I know that, this is a result of me and @daquinons doing the same screen split by two, I didn't want to touch that part because it's been re-designed by @daquinons, and that'll be a concern to solve when the two PR's get merged. Does it make sense?

> I wonder whether it would look better if the "Current Rate..." text here lined up with the "Fetch rates automatically" text above it. This may be more one for @LevinMedia to give feedback on, or something that has been discussed already so feel free to ignore in that case!

Yep, I'm also uncomfortable by that look, but that's how the Figma's are. 7KVvtMQgJxVAUcPPxTB29r-fi-174%3A3922, we should ask @LevinMedia about that part too.

Hope all does make sense.

Reply:
Fixed in 0523c1e. Thanks again!